includes/handle-redirects.php

```diff
@@ -45,6 +45,12 @@ function maybe_do_redirect() {
 
 	$redirect_uri = get_redirect_uri( $request_path );
 
+	// We matched a redirect. Whitelist its domain for wp_safe_redirect().
+	add_filter( 'allowed_redirect_hosts', function( $hosts ) use ( $redirect_uri ) {
```

Comment:
AFAIK this means whatever redirects are inserted get their hosts whitelisted. This goes against common client requirements to limit host whitelisting to a subset of URLs, which are decided on on a per project basis.

This is especially more true since there are no limitations on what users can enter as redirects.

Reply:
The **project** requirement is arbitrary redirects. The plugin cannot allow arbitrary redirects. This is a platform plugin, so we cannot adapt it to the requirements of one specific project. Plenty of projects do want to lock redirects down, so we cannot use this code.

I'm fine with introducing a dedicated custom filter for whitelisting hosts on a per plugin basis. Basically keeping this function, but instead of having logic inside of the  `allowed_redirect_hosts` callback applying a custom filter to the passed `$hosts` argument.

Not sure if I explained it well.